src/main/java/uk/gov/hmcts/reform/em/hrs/job/DeleteOrphanHearingRecordingAatTask.java

```diff
@@ -0,0 +1,139 @@
+
+
+    public DeleteOrphanHearingRecordingAatTask(
+            @Qualifier("jurisdictionCodesContainerClient") BlobContainerClient blobClient,
+                                       HearingRecordingService hearingRecordingService) {
+        this.blobClient = blobClient;
+        this.hearingRecordingService = hearingRecordingService;
+    }
+
+    @Override
+    public void run() {
+
+        logger.info("Started {} job", TASK_NAME);
+
+        StopWatch stopWatch = new StopWatch();
+        stopWatch.start();
+
+        Optional<BlobClient> csvBlobClient = loadWorkbookBlobClient();
+        if (csvBlobClient.isEmpty()) {
+            logger.info("No files present for processing");
+            return;
+        }
+
+        try (ExecutorService executorService = Executors.newFixedThreadPool(defaultThreadLimit);
+             XSSFWorkbook workbook = loadWorkbook(csvBlobClient.get())) {
+
+            XSSFSheet sheet = workbook.getSheetAt(0);
+
+            List<String> unProcessedRecords = new ArrayList<>();
+
+            for (Row row : sheet) {
+                unProcessedRecords.add(getStringCellValue(row.getCell(0)));
+            }
+
+            logger.info("Number of rows in unProcessedRecords: {}", unProcessedRecords.size());
+
+            List<List<String>> batches = Lists.partition(unProcessedRecords, batchSize);
+            logger.info("Number of batches created: {}", batches.size());
+
+            for (List<String> batch : batches) {
+                executorService.submit(() -> processBatch(batch));
+            }
+
+        } catch (IOException e) {
+            logger.info("Encountered error deleting orphan HRS metadata: {}", e.getMessage());
+        }
+
+        csvBlobClient.get().delete();
+        stopWatch.stop();
+
+        logger.info("Deletion job for orphan HRS metadata took {} ms", stopWatch.getDuration().toMillis());
+        logger.info("Finished {} job", TASK_NAME);
+    }
+
+    private void processBatch(List<String> batch) {
+        List<Long> caseIds = batch.stream()
+                .map(Long::valueOf)
+                .toList();
+
+        hearingRecordingService.deleteCaseHearingRecordings(caseIds);
+    }
+
+    private String getStringCellValue(Cell cell) {
+        return Objects.isNull(cell) || cell.getStringCellValue().isBlank() ? null : cell.getStringCellValue();
+    }
+
+    private Optional<BlobClient> loadWorkbookBlobClient() {
+        return blobClient.listBlobs()
+                .stream()
+                .map(blobItem -> blobClient.getBlobClient(blobItem.getName()))
+                .findAny();
+    }
+
+    private XSSFWorkbook loadWorkbook(BlobClient client) throws IOException {
+        File spreadsheetFile = null;
+        try {
+            spreadsheetFile = File.createTempFile("jurisdictionCodes", ".xslx");
+            final String filename = spreadsheetFile.getAbsolutePath();
+
+            Files.deleteIfExists(spreadsheetFile.toPath());
+            client.downloadToFile(filename);
```

Comment:
it deletes file before loading it and  you delete the file in finally anyway, it should be enough?